packages/server/integrations/jiraServer/JiraServerRestManager.ts

```diff
@@ -54,45 +90,103 @@ export default class JiraServerRestManager {
     return json.errorMessages?.join('\n')
   }
 
-  async request(method: string, path: string) {
+  async parseJsonResponse<T>(response: Response): Promise<T | Error> {
+    const contentType = response.headers.get('content-type') || ''
+    if (!contentType.includes('application/json')) {
+      return new Error('Received non-JSON Jira Server Response')
+    }
+    const json = await response.json()
+
+    if (response.status !== 201 && response.status !== 200) {
+      return new Error(
+        `Fetching projects failed with status ${response.status}, ${this.formatError(json)}`
+      )
+    }
+
+    return json
+  }
+
+  async requestRaw(method: string, path: string, body?: any) {
     const url = new URL(path, this.serverBaseUrl)
     const request = {
       url: url.toString(),
       method
     }
     const auth = this.oauth.authorize(request, this.token)
-    const response = await fetch(request.url, {
+    return fetch(request.url, {
       method: request.method,
+      body: body ? JSON.stringify(body) : undefined,
       headers: {
+        'Content-Type': 'application/json',
         ...this.oauth.toHeader(auth)
       }
     })
-    return response
   }
 
-  async parseJsonResponse<T>(response: Response): Promise<T | Error> {
-    const contentType = response.headers.get('content-type') || ''
-    if (!contentType.includes('application/json')) {
-      return new Error('Received non-JSON Jira Server Response')
+  async request<T>(method: string, path: string, body?: any) {
+    const response = await this.requestRaw(method, path, body)
+    return this.parseJsonResponse<T>(response)
+  }
+
+  async getCreateMeta() {
+    return this.request<JiraServerCreateMeta>('GET', '/rest/api/2/issue/createmeta')
+  }
+
+  async getIssue(issueId: string) {
+    return this.request<JiraServerIssue>(
+      'GET',
+      `/rest/api/latest/issue/${issueId}?expand=renderedFields`
+    )
+  }
+
+  async createIssue(projectId: string, summary: string, description: string) {
+    const meta = await this.getCreateMeta()
+    if (meta instanceof Error) {
+      throw meta
```

Comment:
-1 your calling function in the TaskIntegrationManager is expecting the return value to possibly be an Error, but it isn't using a try/catch. all these errors should be returned & not thrown